### ISSUE TITLE
collider build  failed

### ISSUE THREAD

**Issue by dandingol03** [webrtc/apprtc#556]
**Browsers and versions affected**
browser:chrome

**Description**
After I run ``go test collider``,there is some log below:
```
collider
../../go/src/collider/dashboard_test.go:35: Errorf format %d has arg expected of wrong type string
../../go/src/collider/room_test.go:89: Errorf format %v reads arg #4, but call has only 3 args
FAIL	collider [build failed]
```

**Steps to reproduce**
1.go get collidermain
2.go install collidermain
3.generate ssh certificate 
4.run ``go test collider``

**Comment by denniswon** [comment]
Any update on this issue? Having the same prob.

Also, running the server also fails for me using tls:
$GOPATH/bin/collidermain -port=8089 -tls=true

2018/05/20 09:48:35 Starting collider: tls = true, port = 8089, room-server=https://appr.tc
2018/05/20 09:48:35 Run: open /cert/cert.pem: no such file or directory

Could anyone please help? Thanks in advance.

**Comment by deboc** [comment]
Hi, same cast issue here with the Errorf.

@denniswon you probably forgot to create a certificate, like:
```sh
sudo mkdir -p /cert
sudo openssl req -x509 -newkey rsa:2048 -keyout /cert/key.pem -out /cert/cert.pem -days 99999 -nodes
```

**Comment by swifticus2** [comment]
Did anyone resolve this?

**Comment by ivdmitry** [comment]
Hello I have the same error

~$ go test collider
# collider
/src/collider/dashboard_test.go:35: Errorf format %d has arg expected of wrong type string
/src/collider/room_test.go:89: Errorf format %v reads arg #4, but call has only 3 args

the certificate created
collidermain starting work
but on client webrtc 
WebSocket connection to 'wss://test.realmax.fi:8089/ws' failed: Error in connection establishment: net::ERR_CERT_AUTHORITY_INVALID

**Comment by einsteinarbert** [comment]
> Hi, same cast issue here with the Errorf.
> 
> @denniswon you probably forgot to create a certificate, like:
> 
> ```shell
> sudo mkdir -p /cert
> sudo openssl req -x509 -newkey rsa:2048 -keyout /cert/key.pem -out /cert/cert.pem -days 99999 -nodes
> ```

Thank you, I did it, just: sudo chmod -R a+r /cert/

**Comment by EliosSs** [comment]
Hi! excuse me, any update on this error? I am having the same issue.

I already created the certificates and when I run $go test collider

crashes with the message:

collider
../../../../../../Educational/go/src/collider/dashboard_test.go:35:3: Errorf format %d has arg expected of wrong type string
../../../../../../Educational/go/src/collider/room_test.go:89:3: Errorf format %v reads arg #4, but call has 3 args
FAIL	collider [build failed]

**Comment by einsteinarbert** [comment]
i didn't run this test

**Comment by EliosSs** [comment]
but you could use the collider program normally? I mean it is not mandatory for the operation of the program?

**Comment by einsteinarbert** [comment]
Yes, I just did following readme manual page of collinder. But you should
attention about path of project:
[root@ctr486333 goWorkspace]# tree -L 3
.
├── bin
│   └── collidermain
├── cert
│   ├── cert.pem
│   ├── cert.pem_uav
│   └── key.pem
├── cert_uav_autogen
│   ├── cert.pem
│   └── key.pem
├── collider.log
├── nohup.out
├── pkg
│   └── linux_386
│       └── golang.org
├── src
│   ├── collider -> /root/uav-project/apprtc/src/collider/src/collider
│   ├── collidermain ->
/root/uav-project/apprtc/src/collider/src/collidermain
│   ├── collidertest ->
/root/uav-project/apprtc/src/collider/src/collidertest
│   ├── golang.org
│   │   └── x
│   └── nohup.out
└── start.sh
##  in .bashrc:
export GOROOT=/root/uav-project/go
export GOPATH=/root/uav-project/goWorkspace
export GOBIN=$GOROOT/bin
export PATH=$PATH:$GOBIN:$GOPATH



Vào Th 4, 14 thg 8, 2019 vào lúc 09:13 Eliomar Conde <
notifications@github.com> đã viết:

**Comment by einsteinarbert** [comment]
**But now when I deploy apprtc on google cloud, I can connect via chrome of Android without any error but My android native app can't connect, collinder error shown:**
`2019/08/15 16:52:00 http: TLS handshake error from 117.6.160.102:18961: remote error: tls: unknown certificate`
I don't know why because in my server using dev test to run Apprtc, Andriod worked perfect.
Difference is in my server, I used my cert file, but in google cloud I don't use any thing, the domain auto switch to https:
https://my-project-id.appspot.com

**Comment by einsteinarbert** [comment]
cert.pem file is a full chain file (join public and private key into one text file **cert.pem**) :
---- public certificate ----
---- private rsa ------

**Comment by einsteinarbert** [comment]
Dont using google deploy, just using your VPS server, -- tobe continue update late....--

**Comment by wcxdhr** [comment]
> **But now when I deploy apprtc on google cloud, I can connect via chrome of Android without any error but My android native app can't connect, collinder error shown:**
> `2019/08/15 16:52:00 http: TLS handshake error from 117.6.160.102:18961: remote error: tls: unknown certificate`
> I don't know why because in my server using dev test to run Apprtc, Andriod worked perfect.
> Difference is in my server, I used my cert file, but in google cloud I don't use any thing, the domain auto switch to https:
> https://my-project-id.appspot.com

Same issue. I just used these commands to generate certificate. 
```
sudo mkdir -p /cert
sudo openssl req -x509 -newkey rsa:2048 -keyout /cert/key.pem -out /cert/cert.pem -days 99999 -nodes
```
Do you know how to solve this?

**Comment by einsteinarbert** [comment]
just buy a real cert for your server. In Android, we must to use real cert
file.

Vào Th 4, 5 thg 8, 2020 vào lúc 13:21 wcxdhr <notifications@github.com>
đã viết:
